go/api/base_client.go

```diff
@@ -3492,3 +3492,215 @@ func (client *baseClient) XClaimJustIdWithOptions(
 	}
 	return handleStringArrayResponse(result)
 }
+
+// Returns stream entries matching a given range of IDs.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	key   - The key of the stream.
+//	start - The start position.
+//	        Use `options.NewStreamBoundary()` to specify a stream entry ID and its inclusive/exclusive status.
+//	        Use `options.NewInfiniteStreamBoundary()` to specify an infinite stream boundary.
+//	end   - The end position.
+//	        Use `options.NewStreamBoundary()` to specify a stream entry ID and its inclusive/exclusive status.
+//	        Use `options.NewInfiniteStreamBoundary()` to specify an infinite stream boundary.
+//
+// Return value:
+//
+//	A `map` of key to stream entry data, where entry data is an array of
+//	pairings with format `[[field, entry], [field, entry], ...]`. Returns `nil` if `count` is non-positive.
+//
+// Example:
+//
+//	// Retrieve all stream entries
+//	res, err := client.XRange(
+//		"key",
+//		options.NewInfiniteStreamBoundary(options.NegativeInfinity),
+//		options.NewInfiniteStreamBoundary(options.PositiveInfinity),
+//	)
+//	fmt.Println(res) // Output: map[key:[["field1", "entry1"], ["field2", "entry2"]]]
+//
+//	// Retrieve exactly one stream entry by id
+//	res, err := client.XRange(
+//		"key",
+//		options.NewStreamBoundary(streamId, true),
+//		options.NewStreamBoundary(streamId, true),
+//	)
+//	fmt.Println(res) // Output: map[key:[["field1", "entry1"]]
+//
+// [valkey.io]: https://valkey.io/commands/xrange/
+func (client *baseClient) XRange(
```

Comment:
This func is an overload of the `XRangeWithOptions` function differing only by the `opts` parameter. The func should just call the `XRangeWithOptions` func with a `nil` value for the `opt` parameter.